pkg/state/rewards_calculator.go

```diff
@@ -134,6 +141,28 @@ func (c *rewardCalculator) handleFeature21(height proto.Height, rewardAddresses
 	return rewardAddresses
 }
 
+func (c *rewardCalculator) handleFeature23(height proto.Height) (uint64, error) {
+	const (
+		boostedRewardMultiplier = 10
+		defaultRewardMultiplier = 1
+	)
+	// Feature 23 "Boost Block Reward" is working only for `BlockRewardBoostPeriod` count of blocks. We have to check
+	// that feature already activated and not expired yet. In this case the multiplication can be applied, so we return
+	// the value of `boostedRewardMultiplier`.
+	ah, err := c.features.newestActivationHeight(int16(settings.BoostBlockReward))
+	if err != nil {
+		if errors.Is(err, keyvalue.ErrNotFound) { // feature 23 is not approved or activated.
+			return defaultRewardMultiplier, nil
+		}
+		return 0, fmt.Errorf("failed to get activation height for feature 23: %w", err)
+	}
+	boostPeriodLastHeight := ah + c.settings.BlockRewardBoostPeriod
+	if ah > 0 && height >= ah && height < boostPeriodLastHeight {
```

Comment:
Why do you check `ah > 0`? Cat it be 0 or lower that 0?

Reply:
Tests issue, mocks return 0 for inactive features.

Reply:
I think the good way is to return `keyvalue.ErrNotFound` in case when the feature is not enabled. Tests should not affect the code, on my mind.